src/gui/DatabaseTabWidget.cpp

```diff
@@ -798,7 +690,7 @@ void DatabaseTabWidget::modified()
     Database* db = static_cast<Database*>(sender());
     DatabaseManagerStruct& dbStruct = m_dbList[db];
 
-    if (config()->get("AutoSaveAfterEveryChange").toBool() && dbStruct.saveToFilename) {
+    if (config()->get("AutoSaveAfterEveryChange").toBool() && !dbStruct.readOnly) {
         saveDatabase(db);
```

Comment:
I would like an assertion here for `dbWidget->currentMode() != DatabaseWidget::LockedMode` (or perhaps even inside the `saveDatabase()` method.

Reply:
I put one inside ```saveDatabase()``` in place of the interlock.

Reply:
Perhaps you should leave both. The assertion will ensure we notice in debug mode, but the safeguard condition will ensure it never happens in production either.

Reply:
I was thinking that too. Will do!